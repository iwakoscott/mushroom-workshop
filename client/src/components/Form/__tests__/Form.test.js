import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { Form } from '..';
import 'jest-dom/extend-expect';

afterEach(cleanup);

describe(`<Form />`, () => {
  const fakeEmail = 'host@foodnome.com';

  it(`renders without crashing...`, () => {
    const { container } = render(<Form />);
    expect(container).toMatchSnapshot();
  });

  it(`submits a users email`, () => {
    const fakeSubmitHandler = jest.fn();
    const { getByLabelText, getByText } = render(
      <Form onSubmitEmail={fakeSubmitHandler} />
    );

    // DOM Nodes
    const inputNode = getByLabelText(/email/i);
    const submitButtonNode = getByText(/submit/i);

    expect(submitButtonNode).toBeDisabled();

    // change input field
    fireEvent.change(inputNode, { target: { value: fakeEmail } });
    expect(inputNode.value).toBe(fakeEmail);

    // fire submit
    fireEvent.click(submitButtonNode.parentNode);
    expect(fakeSubmitHandler).toHaveBeenCalledTimes(1);
    expect(fakeSubmitHandler).toHaveBeenCalledWith(fakeEmail);
  });
});
