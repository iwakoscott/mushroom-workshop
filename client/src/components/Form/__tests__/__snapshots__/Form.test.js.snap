// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders without crashing... 1`] = `
<div>
  <div
    class="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
    style="width: 100%; max-width: 400px; height: 400px;"
  >
    <form
      style="display: flex; justify-content: space-around; align-items: center; flex-direction: column; height: 100%;"
    >
      <h1>
        Enter your email!
      </h1>
      <div
        class="MuiFormControl-root-29"
      >
        <label
          class="MuiFormLabel-root-44 MuiInputLabel-root-33 MuiInputLabel-formControl-38 MuiInputLabel-animated-41"
          data-shrink="false"
          for="email-input"
        >
          email
        </label>
        <div
          class="MuiInputBase-root-64 MuiInput-root-51 MuiInput-underline-55 MuiInputBase-formControl-65 MuiInput-formControl-52"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-74 MuiInput-input-59 MuiInputBase-inputType-77 MuiInput-inputType-62"
            id="email-input"
            placeholder="email..."
            type="email"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root-107 MuiButtonBase-disabled-108 MuiButton-root-81 MuiButton-text-83 MuiButton-flat-86 MuiButton-disabled-101"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label-82"
        >
          Submit
        </span>
      </button>
    </form>
  </div>
</div>
`;
